pkg/controller/template/template_controller.go

```diff
@@ -171,12 +182,55 @@ func (ctrl *Controller) deleteSecret(obj interface{}) {
 	}
 }
 
+func (ctrl *Controller) enqueueController() {
+	cfg, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get ControllerConfig on dependency callback %#v", err))
+		return
+	}
+	glog.V(4).Infof("Re-syncing ControllerConfig %s due to dependency change", cfg.Name)
+	ctrl.enqueueControllerConfig(cfg)
+}
+
+func (ctrl *Controller) updateFeature(old, cur interface{}) {
+	oldFeature := old.(*osev1.FeatureGate)
+	newFeature := cur.(*osev1.FeatureGate)
+	if !reflect.DeepEqual(oldFeature.Spec, newFeature.Spec) {
+		glog.V(4).Infof("Update Feature %s", newFeature.Name)
+		ctrl.enqueueController()
+	}
+}
+
+func (ctrl *Controller) addFeature(obj interface{}) {
+	features := obj.(*osev1.FeatureGate)
+	glog.V(4).Infof("Adding Feature %s", features.Name)
+	ctrl.enqueueController()
+}
+
+func (ctrl *Controller) deleteFeature(obj interface{}) {
+	features, ok := obj.(*osev1.FeatureGate)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		features, ok = tombstone.Obj.(*osev1.FeatureGate)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a FeatureGate %#v", obj))
+			return
+		}
+	}
+	glog.V(4).Infof("Deleted Feature %s", features.Name)
```

Comment:
```suggestion
	glog.V(4).Infof("Deleting Feature %s", features.Name)
```

It hasn't happened yet right?